app_testing/Makefile

```diff
@@ -24,6 +24,7 @@ APP_SHARED_FLAGS  = -report \
 -Wno-unknown-pragmas \
 -Wno-bidirectional-buffered-port \
 -g \
+-D NUM_THREADS=4 \
```

Comment:
You may remove the space between -D and NUM_THREADS; not sure whether the space always works